### ISSUE TITLE
WCAG - Users not always informed when changes are happening on the map

### ISSUE THREAD

**Issue by jolevesq** [fgpv-vpgf/fgpv-vpgf#3847]
Some users are not always informed that changes made on the map have updated content dynamically on the same page so they will not be aware that the page has changed. E.g., clicking the data table button displays the data table on the same page.

**Comment by jolevesq** [comment]
To add for datatable:

No, this issue has not been fixed. The data table feature in the map viewer generates a table or grid for the text data. But, the data table is not marked up correctly with the proper semantic information necessary to communicate relationships between table header cells and the related data.

The issue in this ticket fails WCAG success criterion 1.3.1: Info and Relationships specifically because the data tables generated by the ramp viewer do not use the appropriate mark up for table headers. A more detailed description of the failure including some techniques for fixing are contained in F91: Failure of Success Criterion 1.3.1 for not correctly marking up table headers .
To fix this, the data tables will have to be created using the proper HTML or ARIA markup for the headers

**Comment by jolevesq** [comment]
Part of this was fix by https://github.com/fgpv-vpgf/fgpv-vpgf/issues/3805 but it looks like there some tag that are still missing

**Comment by dan-bowerman** [comment]
@dan-bowerman to reach out again to Darold Lindquist at NRCan for more direction. 
I think our tables are fine now, but this issue is now just about tracking the announcement of application state changes to a screen reader, e.g. "data grid opened".

**Comment by jolevesq** [comment]
@dan-bowerman Will this be part of 3.3.4? Have you been able to reach out to Darold? If not we can help.

**Comment by dan-bowerman** [comment]
@jolevesq I would appreciate if you could, it has been on my plate to reach out to Darold but I haven't had the opportunity yet. I think we can pretty quickly get a solution together once we have a good idea of what specifically needs to be done. The sooner we have a response, the sooner we can get that issue rolled up into a new release, probably 3.3.4 unless you need the getCapabilities/GeoMet issue fix sooner.

**Comment by dan-bowerman** [comment]
From Darold at NRCan:

> Using aria-live attributes to announce changes on a dynamically updated page is one good way to accomplish this. Messages should be concise with just enough information  to inform users about when the page is updated with the change(s( and what occurred. Maps are even more challenging because information may be updated simultaneously in several different locations on a page.
> With the expand/collapse example, expanding the section with the legend would display the content and interface elements for that. Here, a message of “Expanded” and “Collapsed” may be sufficient. 
> Reading the newly displayed content and using the interface controls is a related but separate interaction.
> With the data table example, using the generate data table button in the map interface should insert a properly marked up data table on the page. In this case, inserting the message “data table loaded” when the data table is displayed would be sufficient.
> In some, but not all cases, moving the focus to updated content on the page could also be useful. 
> Keep in mind that in the case of OSDP the map is loaded in an iframe. Although this is a technically correct technique, it is known to produce inconsistent and unexpected results for screen readers.

**Comment by spencerwahl** [comment]
Possible implementation idea;

An alert system that inserts text into an invisible element with `role="alert"`, this will allow us to alert screen reader users with messages about whats happening on the screen. For instance when the data table is opened the text "Data table loaded" is inserted into the element and the screen reader will read it as an alert. I'm unsure if just changing the text of the element will work or if we'd have to remake the element every time.
